example/rest-api/art-institute/art-institute.php

```diff
@@ -31,7 +31,32 @@ function register_aic_block(): void {
 	$get_art_query = HttpQuery::from_array([
 		'data_source' => $aic_data_source,
 		'endpoint' => function ( array $input_variables ) use ( $aic_data_source ): string {
-			return sprintf( '%s/%s', $aic_data_source->get_endpoint(), $input_variables['id'] ?? '' );
+			$endpoint = $aic_data_source->get_endpoint();
+			$ids = [];
+				
+			if ( isset( $input_variables['id'] ) ) {
+				$ids[] = $input_variables['id'];
+			} else {
+				foreach ( $input_variables as $input ) {
```

Comment:
This seems a bit complicated, what case does this cover?

Reply:
Ya 😬 This is to ensure that a single item can still be selected and transforms it into an array to handle either a single item or multi-item selection.

In the code, it first checks if an 'id' key and if it's there, it creates a single-item array. Otherwise, it filters the array of input variables for items with 'id' and maps them to extract the id values. 

I added comments and tried to make the code easier to read, but not sure if I've achieved it. If you have any suggetions, please share! 

https://github.com/Automattic/remote-data-blocks/blob/8ca83120bce3d91fc4238816d2464fd96bba4ada/example/rest-api/art-institute/art-institute.php#L35-L43